File: problems-extend-adapt/3801-3900/3813_vowels-per-consonant/solution.go
func vowelsPerConsonant(s string) int {
	// One pass tallies both totals: each character either is one of
	// the five vowels and bumps v, is another lowercase letter and
	// bumps c, or is a space or digit and bumps neither. The score is
	// then the integer quotient floor(v / c), or 0 when no consonant
	// exists to divide by.
	v := 0
	c := 0
	for _, ch := range s {
		if ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' {
			v++
		} else if ch >= 'a' && ch <= 'z' {
			c++
		}
	}
	if c == 0 {
		return 0
	}
	return v / c
}

File: problems-extend-adapt/3801-3900/3813_vowels-per-consonant/solution.py
from typing import List, Optional


class Solution:
    def vowelsPerConsonant(self, s: str) -> int:
        # One pass tallies both totals: each character either is one of
        # the five vowels and bumps v, is another lowercase letter and
        # bumps c, or is a space or digit and bumps neither. The score is
        # then the integer quotient floor(v / c), or 0 when no consonant
        # exists to divide by.
        v = 0
        c = 0
        for ch in s:
            if ch in "aeiou":
                v += 1
            elif "a" <= ch <= "z":
                c += 1
        return v // c if c else 0
